libcontainerd/remote/client.go

```diff
@@ -711,6 +711,35 @@ func (c *client) processEvent(ctx context.Context, et libcontainerdtypes.EventTy
 	})
 }
 
+func (c *client) waitServe(ctx context.Context) bool {
+	t := 100 * time.Millisecond
+	delay := time.NewTimer(t)
+	if !delay.Stop() {
+		<-delay.C
+	}
+	defer delay.Stop()
+
+	// `IsServing` will actually block until the service is ready.
+	// However it can return early, so we'll loop with a delay to handle it.
+	for {
+		serving, _ := c.client.IsServing(ctx)
+		if ctx.Err() != nil {
+			return false
+		}
```

Comment:
Wondering if if would be useful to return the error (and log it?) instead of returning a boolean; so `err == nil` means "yup, it's up now", and `err != nil` means something went wrong

Reply:
@cpuguy83 ^

Reply:
I can do a better job at checking the error from `IsServing` and logging cases that are not context related.

I don't think returning an error here is what we want though, since this should loop forever until the condition is true or the context is cancelled.

Reply:
Updated this.